File: OsmAnd/src/net/osmand/plus/settings/vehiclespecs/SpecificationType.java
package net.osmand.plus.settings.vehiclespecs;

import static net.osmand.router.GeneralRouter.MAX_AXLE_LOAD;
import static net.osmand.router.GeneralRouter.VEHICLE_HEIGHT;
import static net.osmand.router.GeneralRouter.VEHICLE_LENGTH;
import static net.osmand.router.GeneralRouter.VEHICLE_WEIGHT;
import static net.osmand.router.GeneralRouter.VEHICLE_WIDTH;
import static net.osmand.router.GeneralRouter.WEIGHT_RATING;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.osmand.util.CollectionUtils;

public enum SpecificationType {

	WIDTH(VEHICLE_WIDTH),
	HEIGHT(VEHICLE_HEIGHT),
	LENGTH(VEHICLE_LENGTH),
	WEIGHT(VEHICLE_WEIGHT),
	AXLE_LOAD(MAX_AXLE_LOAD),
	WEIGHT_FULL_LOAD(WEIGHT_RATING);

	private final String key;

	SpecificationType(@NonNull String key) {
		this.key = key;
	}

	@Nullable
	public static SpecificationType getByKey(@NonNull String key) {
		for (SpecificationType type : values()) {
			if (key.equals(type.key)) {
				return type;
			}
		}
		return null;
	}

	public boolean isWeightRelated() {
		return CollectionUtils.equalsToAny(this, WEIGHT, AXLE_LOAD, WEIGHT_FULL_LOAD);
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/vehicle/specification/domain/SpecificationType.kt
package net.osmand.shared.vehicle.specification.domain

import kotlin.jvm.JvmStatic

enum class SpecificationType(val key: String) {

	WIDTH("width"),
	HEIGHT("height"),
	LENGTH("length"),
	WEIGHT("weight"),
	AXLE_LOAD("maxaxleload"),
	WEIGHT_FULL_LOAD("weightrating");

	companion object {

		@JvmStatic
		fun getByKey(key: String): SpecificationType? {
			return entries.find { it.key == key }
		}
	}

	fun isWeightRelated(): Boolean {
		return when (this) {
			WEIGHT, AXLE_LOAD, WEIGHT_FULL_LOAD -> true
			else -> false
		}
	}
}